apps/web/src/views/Swap/V3Swap/containers/SwapCommitButton.tsx

```diff
@@ -81,7 +82,14 @@ export const SwapCommitButton = memo(function SwapCommitButton({
   const showWrap = wrapType !== WrapType.NOT_APPLICABLE
   const [isRoutingSettingChange, resetRoutingSetting] = useRoutingSettingChanged()
   const slippageAdjustedAmounts = useSlippageAdjustedAmounts(trade)
-  const routerAddress = SMART_ROUTER_ADDRESSES[trade?.inputAmount?.currency?.chainId]
+
+  const deadline = useTransactionDeadline()
+  const [statusWallchain, approvalAddressForWallchain, wallchainMasterInput] = useWallchainApi(trade, deadline)
+  const [wallchainSecondaryStatus, setWallchainSecondaryStatus] = useState('not-found')
```

Comment:
```suggestion
  const [wallchainSecondaryStatus, setWallchainSecondaryStatus] = useState<'found'|'not-found'>('not-found')
```

why not just use Boolean